src/main/java/study/step2/StringAddCalculator.java

```diff
@@ -0,0 +1,63 @@
+package study.step2;
+
+
+import java.util.Arrays;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class StringAddCalculator {
+
+    private static final String DEFAULT_DELIMITER;
+    private static final String CUSTOM_DELIMITER;
+    private static final Pattern CUSTOM_DELIMITER_PATTERN;
+
+    static {
+        DEFAULT_DELIMITER = ":|,";
+        CUSTOM_DELIMITER = "//(.*)\n(.*)";
+        CUSTOM_DELIMITER_PATTERN = Pattern.compile(CUSTOM_DELIMITER);
+    }
+
+    public static int calculate(String input) {
+        if (isEmptyOrNull(input)) return 0;
+
+        Matcher matcher = CUSTOM_DELIMITER_PATTERN.matcher(input);
+        if (matcher.find()) {
+            String customDelimiter = matcher.group(1);
+            String targetString = matcher.group(2);
+            return sum(splitString(targetString, customDelimiter));
+        }
+        
+        String[] split = splitString(input, DEFAULT_DELIMITER);
+        if (split.length > 1) {
+            return sum(split);
+        }
```

Comment:
matcher가 되던 안되던, 결국 sum() 를 호출하고 있군요!
`findDelimiter()` 메서드를 만들어서 리팩토링이 가능해보입니다!

```java
private static String findDelimiter(String text) {
    Matcher m = CUSTOM_DELIMITER_PATTERN.matcher(text); 
    if (m.find()) { 
        return DEFAULT_DELIMITER + '|' + m.group(1); 
    } 
    return DEFAULT_DELIMITER; 
}
```

```suggestion
        String delimiter = findDelimiter(input);
        String[] split = input.split(delemiter);
	return sum(split);
```

Reply:
메서드를 분리하는데 도움이 많이 되었습니다.
findDelimiter()에서 custom delimiter (m.group(1)) 만 추출하고 끝내기에는 너무 아까워서
target이되는 string(m.group(2))을 구해 split 까지 처리하도록 하고, 함수명을 split()으로 변경했습니다.